### ISSUE TITLE
Detail: Add content on How we prevent MITM attacks

### ISSUE THREAD

**Issue by enclave-alistair** [enclave-networks/docs#49]
Capture content from slack conversation:

```
When a connection is established between Alice and Bob, we perform a mutual authentication handshake that requires proof-of-possession on both sides.

Specifically, during the handshake, each peer sends its signed certificate to the other; this certificate contains the peer's public key.  Certificates look something like this:

"Certificate": {
    "Version": 0,
    "SerialNumber": "n7JLyhs/mUivvK1ALOnxjA==",
    "PermittedUse": "Endpoint",
    "SubjectDistinguishedName": "R8WLQ",
    "SubjectPublicKey": "GwkfXjwAqxJheLzVWhF+V8JEa0R9od74ysQfC6i2Yas=",
    "NotBefore": 1673538715,
    "NotAfter": -1,
    "IssuerDistinguishedName": "Enclave Networks Primary Intermediate G1",
    "IssuerPublicKey": "WwZg4p39Swi3J/z/XfqWPV1Ld466U0VVr11x8bclPOs=",
    "Signature": "rVEmjvZUbiaa7SSnfdYPhV87tGWI1B3Sdrn2kU2Ir/Wj2XSdId1IwxAi3GXgRPEbH2LPlH8ysWe5jLvhLf99Dg=="
}

During the handshake, peers are required to sign a generated 128-bit challenge value with their private key (signing performed with the Ed25519 algorithm); the other peer then verifies that signature matches using the public key in the certificate. In that way we have ensured proof-of-possession.

In addition to proof-of-possession, both sides also verify that the provided signature of the peer's certificate is valid, and the issuer is our known and trusted issuing authority. In this way we guarantee that the certificate has not been fabricated.

Finally, once we have confirmed that the certificates on both sides are valid, and we have proof-of-possession, we derive new secret symmetric per-session keys for the connection using the Curve25519 key agreement agreement algorithm, and all subsequent data transmission is encrypted those keys.

This pattern results in an end-to-end encrypted data stream that no-one except the two peers are able to decrypt.

So, to sum it up, for a successful man-in-the-middle attack to occur, they would need to either:

- Exfiltrate the stored private key from one of the two peers (we take measures on each platform to reduce the chances of this).
- Exfiltrate our own issuer private key from our servers to issue a valid certificate to themselves (we are very careful with our platform keys).
- Somehow gain knowledge of the temporary per-session secret keys from each peer (very difficult, and we take steps to prevent in-memory key material from being accessed via core dumps).


```
